packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -502,6 +486,22 @@ export async function parseRequestObject(requestObject: IRequestOptions) {
 	}
 	axiosConfig.httpsAgent = new Agent(agentOptions);
 
+	axiosConfig.beforeRedirect = (redirectedRequest) => {
```

Comment:
The two code blocks are identical. Can we extract this into a function like this
```typescript
const getBeforeRedirectFn =
	(agentOptions: AgentOptions, axiosConfig: AxiosRequestConfig<any>) =>
	(redirectedRequest: Record<string, any>) => {
		const redirectAgent = new Agent({
			...agentOptions,
			servername: redirectedRequest.hostname,
		});
		redirectedRequest.agent = redirectAgent;
		redirectedRequest.agents.https = redirectAgent;

		if (axiosConfig.headers?.Authorization) {
			redirectedRequest.headers.Authorization = axiosConfig.headers.Authorization;
		}
		if (axiosConfig.auth) {
			redirectedRequest.auth = `${axiosConfig.auth.username}:${axiosConfig.auth.password}`;
		}
	};
```